### DISCUSSION TITLE
Primevue Locale

### DISCUSSION THREAD

**Discussion by Andrek18** [primefaces/community#845] [PrimeVue] [answered]
I use PrimVue (Nuxt-Version) and want to use the german language. I add the PrimeVue Local (https://github.com/primefaces/primelocale) to my package.json. But now.. i do not know how i can set the language in the options or bett from the language of the browser/os. There is a short Information about the lang on https://primevue.org/configuration/#locale. But it did not help me how to do it.

**Comment by melloware** [comment, marked as answer]
Here is a great PrimeVue showcase showing switching locales: https://codesandbox.io/p/devbox/primevue-locale-loader-gxm2ct

**Reply by Andrek18** [reply]
Thanks for the showcase. I copied it. But when ein used it in my app.vue browserLocal and osLocal has errors at runtime. And when i set "de" i get an "Cannot convert undefined or null to object" at the primevue.config.locale. I think the files not located at runtime. I have primlocal in my node_models and the files are there. I think it could be a problem with nuxt and the server side rendering. But i am new at this. Do you have any idea?

**Reply by melloware** [reply]
If you create a StackBlitz showing your issue I am sure someone here can help look at it.

**Reply by moaabid** [reply]
@melloware Arabic values are not getting changed in the example you have provided.  Model value is getting changed but not the  text in input field
<img width="1440" alt="Screenshot 2024-01-15 at 10 53 37 PM" src="https://github.com/primefaces/community/assets/42162174/84924838-6723-48a5-b5d0-5633167a4f09">

**Reply by melloware** [reply]
@moaabid i would open a PrimeVue ticket on gitHub then since its is most likely PrimeVue related.

**Comment by rui-filipe-ribeiro** [comment]
Some PrimeVue information is scarce. But try this:
First put the de.ts file in ./locale folder
In the main.ts file:
```
import { de } from "./locale/de"

const app = createApp(App);
app.use(PrimeVue, { locale: de });
```

**Reply by Andrek18** [reply]
Frim primelocals i have a de.json but no de.ts. Where can i found an de.ts?

**Reply by rui-filipe-ribeiro** [reply]
In my case, I changed pt.json to pt.ts, exporting the json values:

```
export const pt = {
"var": "value",
...
}
```

**Reply by Andrek18** [reply]
Sry for the late replay. @rui-filipe-ribeiro  your suggestions works thanks
